File: modern-java/UC-BerkeleyCS61B-Data-Structures/code/week2/code/worksheet/Planet.py
from __future__ import annotations
import math

class Planet:
    x: float
    y: float
    mass: float

    def __init__(self,x,y,mass):
        self.x = x
        self.y = y
        self.mass = mass
        
    def distance_to(self,other: Planet):
        return math.sqrt(
            (other.x - self.x) **2 + (other.y - self.y)**2
        )
        
    @staticmethod
    def total_mass(planets: list[Planet]):
        total = 0
        for p in planets:
            total += p.mass
        return total
            
p1 = Planet(5,10,100)
p2 = Planet(1,2,200)
print(p1.distance_to(p2))
print(Planet.total_mass([p1,p2]))

File: modern-java/UC-BerkeleyCS61B-Data-Structures/code/week2/code/worksheet/Planet.java
import java.util.List;

///usr/bin/env jbang "$0" "$@" ; exit $?
//JAVA 25+

public class Planet {
	double x;
	double y;
	double mass;

	public Planet(double x, double y, double mass) {
		this.x = x;
		this.y = y;
		this.mass = mass;
	}

	public double distanceTo(Planet other) {
		double dx = other.x - this.x;
        double dy = other.y - this.y;
        return Math.sqrt(dx * dx + dy * dy);
    }

	public static double totalMass(List<Planet> planets) {
		return planets.stream()
			.mapToDouble(p -> p.mass)
			.sum();

	}
}

void main(String... args) {
    var p1 = new Planet(5, 10, 100);
    var p2 = new Planet(1, 2, 200);
    IO.println(p1.distanceTo(p2));
    IO.println(Planet.totalMass(List.of(p1, p2)));
}